core/src/main/scala/scalaz/Generator.scala

```diff
@@ -1,29 +1,32 @@
 package scalaz
 
 /**
- * A `Generator[C]` is a container of elements, and which knows how to efficiently apply a [[scalaz.Reducer]]
+ * A `Generator[C, Elem]` is a container of elements, and which knows how to efficiently apply a [[scalaz.Reducer]]
```

Comment:
Seems like `C` is the container (not `Generator[C, Elem]`)

Reply:
The main purpose of the Generator/Reducer classes is to be a lightweight general purpose map-reduce framework. So the Generator must have the ability to reduce in parallel, aggregating from both sides. Caller is responsible to chose Generator/reducer pair that work well with each other.

Reply:
Well, currently Generator is not a type class, but Reducer is one, and probably need to stay so, to ensure coherence with Semigroup.